pyfolio/perf_attrib.py

```diff
@@ -112,27 +124,97 @@ def create_perf_attrib_stats(perf_attrib):
     common_returns = perf_attrib['common_returns']
 
     summary['Annual multi-factor alpha'] =\
-        empyrical.annual_return(specific_returns)
+        ep.annual_return(specific_returns)
 
     summary['Multi-factor sharpe'] =\
-        empyrical.sharpe_ratio(specific_returns)
+        ep.sharpe_ratio(specific_returns)
 
     summary['Cumulative specific returns'] =\
-        empyrical.cum_returns(specific_returns)
+        ep.cum_returns_final(specific_returns)
     summary['Cumulative common returns'] =\
-        empyrical.cum_returns(common_returns)
+        ep.cum_returns_final(common_returns)
     summary['Total returns'] =\
-        empyrical.cum_returns(perf_attrib['total_returns'])
+        ep.cum_returns_final(perf_attrib['total_returns'])
 
     summary = pd.Series(summary)
     return summary
 
 
-def show_perf_attrib_stats(perf_attrib_data, risk_exposures):
+def show_perf_attrib_stats(returns, positions, factor_returns,
+                           factor_loadings):
     """
-    Takes perf attribution data over a period of time, computes stats on it,
-    and displays them using `utils.print_table`.
+    Calls `perf_attrib` using inputs, and displays outputs using
+    `utils.print_table`.
     """
+    risk_exposures, perf_attrib_data = perf_attrib(returns,
+                                                   positions,
+                                                   factor_returns,
+                                                   factor_loadings)
+
     perf_attrib_stats = create_perf_attrib_stats(perf_attrib_data)
     print_table(perf_attrib_stats)
     print_table(risk_exposures)
+
+
+def plot_returns(returns, specific_returns, common_returns, ax=None):
+    """
+    Plot total, specific, and common returns.
+    """
+    if ax is None:
+        ax = plt.gca()
+
+    ax.plot(returns, color='g', label='Total returns')
+    ax.plot(specific_returns, color='b', label='Cumulative specific returns')
+    ax.plot(common_returns, color='r', label='Cumulative common returns')
+
+    ax.set_title('Time Series of cumulative returns')
+    ax.set_ylabel('Returns')
+    ax.legend()
+
+    return ax
+
+
+def plot_alpha_returns(alpha_returns, ax=None):
+    """
+    Plot histogram of daily multi-factor alpha returns.
+    """
+    if ax is None:
+        ax = plt.gca()
+
+    ax.hist(alpha_returns, color='g', label='Multi-factor alpha')
+    ax.set_title('Histogram of alphas')
+    ax.axvline(0, color='k', linestyle='--', label='Zero')
+
+    avg = alpha_returns.mean()
+    ax.axvline(avg, color='b', label='Mean = {: 0.5f}'.format(avg))
+    ax.legend()
+
+    return ax
+
+
+def plot_factor_contribution_to_perf(exposures, perf_attrib_data, ax=None):
+    """
+    Plot each factor's contribution to performance.
+
+    Parameters
+    ----------
+    perf_attrib_data : pd.DataFrame
+        df with factors, common returns, and specific returns as columns,
```

Comment:
Need an example dataframe of what this should look like.

Reply:
fixed